tests/e2e/testcases/icm/deploy/suite.go

```diff
@@ -0,0 +1,441 @@
+			gomega.Expect(output).Should(gomega.ContainSubstring("ICM Messenger has already been deployed to C-Chain"))
+			gomega.Expect(output).Should(gomega.ContainSubstring("ICM Registry successfully deployed to C-Chain"))
+		})
+
+		ginkgo.It("should deploy ICM registry with messenger not deployed", func() {
+			globalFlags := utils.GlobalFlags{
+				"local":             true,
+				"skip-update-check": true,
+			}
+			testFlags := utils.TestFlags{
+				"deploy-messenger": "false",
+				"key":              ewoqKeyName,
+			}
+			commandArguments := []string{
+				"--c-chain",
+			}
+
+			_, err := utils.TestCommand(utils.ICMCmd, "deploy", commandArguments, globalFlags, testFlags)
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			output, err := utils.TestCommand(utils.ICMCmd, "deploy", commandArguments, globalFlags, testFlags)
+			gomega.Expect(err).Should(gomega.BeNil())
+			gomega.Expect(output).ShouldNot(gomega.ContainSubstring("ICM Messenger has already been deployed to C-Chain"))
+			gomega.Expect(output).ShouldNot(gomega.ContainSubstring("ICM Messenger successfully deployed to C-Chain"))
+			gomega.Expect(output).Should(gomega.ContainSubstring("ICM Registry successfully deployed to C-Chain"))
+		})
+
+		ginkgo.It("should force deploy ICM registry with messenger not deployed", func() {
+			globalFlags := utils.GlobalFlags{
+				"local":             true,
+				"skip-update-check": true,
+			}
+			testFlags := utils.TestFlags{
+				"deploy-messenger": "false",
+				"key":              ewoqKeyName,
+			}
+			commandArguments := []string{
+				"--c-chain",
+			}
+
+			_, err := utils.TestCommand(utils.ICMCmd, "deploy", commandArguments, globalFlags, testFlags)
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			output, err := utils.TestCommand(utils.ICMCmd, "deploy", append(commandArguments, "--force-registry-deploy"), globalFlags, testFlags)
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			gomega.Expect(err).Should(gomega.BeNil())
+			gomega.Expect(output).ShouldNot(gomega.ContainSubstring("ICM Messenger has already been deployed to C-Chain"))
+			gomega.Expect(output).ShouldNot(gomega.ContainSubstring("ICM Messenger successfully deployed to C-Chain"))
+			gomega.Expect(output).Should(gomega.ContainSubstring("ICM Registry successfully deployed to C-Chain"))
+		})
+
+		ginkgo.It("should deploy ICM messenger and force deploy registry", func() {
+			globalFlags := utils.GlobalFlags{
+				"local":             true,
+				"skip-update-check": true,
+			}
+			testFlags := utils.TestFlags{
+				"deploy-messenger": "false",
+				"key":              ewoqKeyName,
+			}
+			commandArguments := []string{
+				"--c-chain",
+			}
+
+			_, err := utils.TestCommand(utils.ICMCmd, "deploy", commandArguments, globalFlags, testFlags)
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			output, err := utils.TestCommand(utils.ICMCmd, "deploy", append(commandArguments, "--force-registry-deploy"), globalFlags, utils.TestFlags{
```

Comment:
@felipemadero do we need this flag --force-registry-deploy? Why dont we always deploy registry even if messenger is already deployed by default?

Reply:
by default it tries to avoid deploying a registry if there is a registry already deployed. the command can be improved
I believe we can make a ticket for this. ideally we should be able to find out if a registry is present, we don't 
have yet a way to do this, we can only check if a messenger is deployed.

Reply:
that is, by default it will assume a registry was deployed if a messenger is present. but there are cases where
the messenger is there but the registry not, so you need to force the deploy. open to suggestions in appropriate ticket based on this comments